tv_apps/smart-home/js/home.js

```diff
@@ -113,9 +114,48 @@
           // appearance is the same as it.
           that.searchButton.classList.add('hidden');
         }.bind(that));
+
+        // handle animation
+        that.endBubble = null;
+        document.addEventListener(
+                'visibilitychange', that.onVisibilityChange.bind(that));
+        if (document.visibilityState === 'visible') {
+          that.onVisibilityChange();
+        }
       });
     },
 
+    onVisibilityChange: function() {
```

Comment:
We should write more comments here to tell the why we need to force end the animation and the relationship between these animations.